conanfile.py

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import os.path
+from conans import ConanFile, CMake
+from conans.tools import Git
+from conans.errors import ConanInvalidConfiguration
+
+class ImguiSfmlConan(ConanFile):
+    """ Imgui-SFML library conan file.
+
+    Options:
+
+    * shared: True or False
+      Whether to build this library shared or static
+
+    * fPIC: True or False
+      Whether or not to use -fPIC option while building
+
+    * imconfig: None or String
+      Use 'None' if you want this library to provide default
+      ImGui's imconfig.h file. Otherwise, provide a path to the
+      imconfig.h file you want to include in the build.
+
+    * imconfig_install_folder: None or String
+      Use 'None' if you want this build to include imconfig.h file,
+      otherwise use anything else.
+
+    * imgui_revision: String
+      Tag or branch of ImGui repository that should be used with
+      ImGui-SFML.
+    """
+
+    name = 'imgui-sfml'
+    version = '2.0.1'
+    description = 'ImGui binding for use with SFML'
+    topics = ('conan', 'sfml', 'gui', 'imgui')
+    url = 'https://github.com/eliasdaler/imgui-sfml'
+    homepage = 'https://github.com/eliasdaler/imgui-sfml'
+    author = 'Elias Daler <eliasdaler@yandex.ru>'
+    requires = 'sfml/2.5.1@bincrafters/stable'
+    license = 'MIT'
+    settings = 'os', 'compiler', 'build_type', 'arch'
+    options = {
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'imconfig': 'ANY',
+        'imconfig_install_folder': 'ANY',
+        'imgui_revision': 'ANY'
+    }
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'imconfig': None,
+        'imconfig_install_folder': None,
+        'imgui_revision': 'v1.70'
+    }
+    exports_sources = [
+        'cmake/FindImGui.cmake'
+        'CMakeLists.txt',
+        'imconfig-SFML.h',
+        'imgui-SFML.cpp',
+        'imgui-SFML_export.h',
+        'imgui-SFML.h',
+    ]
+    exports = 'LICENSE.md'
+    _imgui_dir = 'imgui'
+
+    def source(self):
+        git = Git(folder=self._imgui_dir)
+        git.clone('https://github.com/ocornut/imgui.git', str(self.options.imgui_revision))
+
+    def configure(self):
+        imconfig = self.options.imconfig
+        if imconfig and not os.path.isfile(str(imconfig)):
```

Comment:
This makes me think... can we handle relative paths somehow? E.g. if I have `~/imgui-sfml/custom.h`, I want to run 
```sh
conan . create imgui-sfml/testing -o imconfig=custom.h
``` 

or at least 

```sh
conan . create imgui-sfml/testing -o imconfig=./custom.h
```

Reply:
I just tried that in the repository folder, and it worked. 
It also worked outside of the repository folder. So, relative paths work.

Reply:
Strange, it didn't work for me. I think it depends where you build. If the current folder when you run "configure" is the same as your current folder, then yes, it will work.
But if you are building somewhere else, it won't. I'll test it again in a few hours.

Reply:
I will also try to verify this behavior in some corner cases. Because it seems very tricky.

Reply:
Alright. I did find the error and fixed it, but the file won't get installed. I also might fix that in conan, though, I just use CMake install -- this means that plain build won't include specified file. 

Is this ok ?

Reply:
Okay, so here's how custom imconfig should be used.
If IMGUI_SFML_CONFIG_NAME is set, then IMGUI_SFML_CONFIG_DIR should be set to where this header can be found.
If IMGUI_SFML_CONFIG_INSTALL_DIR is not set, then this custom config will be installed with ImGui-SFML, otherwise it won't be installed, but it will use IMGUI_SFML_CONFIG_INSTALL_DIR, so when you link to ImGi-SFML CMake target, you'll get IMGUI_SFML_CONFIG_INSTALL_DIR in your include directories list.

So, imagine this. You have "~/imgui-sfml/my-config.h".
If you set this:

```
* IMGUI_SFML_USE_DEFAULT_CONFIG = OFF
* IMGUI_SFML_USER_CONFIG_DIR = ~/imgui-sfml
* IMGUI_SFML_USER_CONFIG_NAME = "my-config.h"
* IMGUI_SFML_CONFIG_INSTALL_DIR=~/my-custom-imgui-sfml/include
```

Then you'll get this:

1) During ImGui-SFML build, "~/imgui-sfml/my-config.h" will be used for building, but it won't be installed
2) If you link to ImGui-SFML, you'll get  "~/my-custom-imgui-sfml/include" to your include directories, because that's where "myconfig.h" will be when you install everything.

This is **very** complicated case, but I hope it's clear now.

Reply:
So, here are the cases which I would like to see working (taking example further)
Suppose that pwd=~/imgui-sfml.

1) ```conan . create imgui-sfml/testing -o imconfig=my-config.h``` - this will us set:

```
* IMGUI_SFML_USE_DEFAULT_CONFIG = OFF
* IMGUI_SFML_USER_CONFIG_DIR = ~/imgui-sfml
* IMGUI_SFML_USER_CONFIG_NAME = "my-config.h"
```

```conan . create imgui-sfml/testing -o imconfig=./my-config.h``` should do the same

2) ```conan . create imgui-sfml/testing -o imconfig=my-config.h -o imconfig_install_folder=~/my-custom-imgui-sfml/include``` - this will us set:

```
* IMGUI_SFML_USE_DEFAULT_CONFIG = OFF
* IMGUI_SFML_USER_CONFIG_DIR = ~/imgui-sfml
* IMGUI_SFML_USER_CONFIG_NAME = "my-config.h"
* IMGUI_SFML_CONFIG_INSTALL_DIR= ~/my-custom-imgui-sfml/include
```

3) ```conan . create imgui-sfml/testing -o imconfig=~/custom-config/my-config.h -o imconfig_install_folder=~/my-custom-imgui-sfml/include``` - this will set:

```
* IMGUI_SFML_USE_DEFAULT_CONFIG = OFF
* IMGUI_SFML_USER_CONFIG_DIR = ~/custom-config
* IMGUI_SFML_USER_CONFIG_NAME = "my-config.h"
* IMGUI_SFML_CONFIG_INSTALL_DIR= ~/my-custom-imgui-sfml/include
```